src/main/java/pl/futurecollars/invoicing/db/file/FileBasedDatabase.java

```diff
@@ -12,17 +12,22 @@
 
 public class FileBasedDatabase implements Database {
 
-  private final Path databasePath;
   private final FilesService filesService;
   private final JsonService jsonService;
   private final IdService idService;
 
+  private final Path databasePath;
+  private final Path idPath;
+
   public FileBasedDatabase(FilesService filesService, JsonService jsonService, IdService idService) throws IOException {
     this.filesService = filesService;
     this.jsonService = jsonService;
     this.idService = idService;
-    databasePath = Path.of(Configuration.DATABASE_FILE);
-    filesService.createRepository(Path.of(Configuration.ID_FILE), databasePath);
+    this.databasePath = Path.of(Configuration.DATABASE_FILE);
+    this.idPath = Path.of(Configuration.ID_FILE);
+    filesService.initDatabase(idPath, databasePath);
```

Comment:
fileService nie powinien mieć metody initDatabase, ponieważ dlaczego fileService, tzn. coś co zajmuje się obsługą plików miałoby mieć metodę tworzącą bazę danych? Trochę to nie pasuje.

Reply:
Ale to nadal kwestia nazewnictwa

Reply:
zarówno metody jak i jej argumentów

Reply:
Wystarczyłoby wywalić tę metodę i w basie danych korzystać bezpośrednio z createFile(...).
Oczywiście dobrze by było wynieść to 2 x createFile do osobnej metody prywatnej w bazie.